### PR TITLE
Associativity of pullbacks

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Proves $(A ×_X B) ×_Y C ≃ A ×_X (B ×_Y C)$ for standard and non-standard pullbacks.

### CODE REVIEW

File: src/foundation-core/pullbacks.lagda.md [hunk 1]
```diff
@@ -502,6 +502,101 @@ module _
...
+```
+
+and with pullback cones `α` and `β` respectively. Moreover, consider a cone `γ`
+over the pullbacks as in the following diagram
+
+```text
+  ∙ ------------> ∙ ------------> C
+  |               | ⌟             |
+  |       γ       |       β       | i
+  |               ∨               ∨
```
Comment (EgbertRijke) [inline src/foundation-core/pullbacks.lagda.md:521]: Should this be

```suggestion
  ∨               ∨               ∨
```

?

It looks like your central dots are not terminal objects here, and that this map is not an identifications.

File: src/foundation/standard-pullbacks.lagda.md [hunk 2]
```diff
@@ -271,6 +271,188 @@ triangle-map-commutative-standard-pullback :
...
+
+```text
+      f       g       g       h
+  A ----> X <---- B ----> X <---- C.
+```
+
+- See also the following relevant stack exchange question:
+  [Assocaitivity of pullbacks](https://math.stackexchange.com/questions/2046276/associativity-of-pullbacks).
+
+#### The standard ternary pullback
```
Comment (EgbertRijke) [inline src/foundation/standard-pullbacks.lagda.md:340]: We could place this under `## Definitions`. I don't think people looking for the standard ternary pullback would necessarily be looking for it here.

**Review by EgbertRijke — APPROVED** [review]
Great pull request!

File: src/foundation-core/pullbacks.lagda.md (hunk 1 above)
Comment (fredrik-bakke) [reply src/foundation-core/pullbacks.lagda.md:521]: Thanks! The central dots are meant to be unnamed vertices. Do you find the style confusing, or was the comment mainly to justify your question?

File: src/foundation/standard-pullbacks.lagda.md (hunk 2 above)
Comment (fredrik-bakke) [reply src/foundation/standard-pullbacks.lagda.md:340]: Hmm. In my mind it fits better here. I see it as similar to how we define types of homotopies where we also characterize the identity types, as opposed to as an unmotivated definition in the `## Definitions` section

Comment (fredrik-bakke) [reply src/foundation/standard-pullbacks.lagda.md]: Never mind, I changed my mind 😅

Comment (fredrik-bakke) [reply src/foundation/standard-pullbacks.lagda.md]: One could probably argue that the standard ternary pullback should be in a new file, but then associativity will be kinda out of place

File: src/foundation-core/pullbacks.lagda.md (hunk 1 above)
Comment (EgbertRijke) [reply src/foundation-core/pullbacks.lagda.md]: No, the style is not confusing. Perhaps you could draw an extra pullback-corner in the top-left square just to be more explicit

Comment (fredrik-bakke) [reply src/foundation-core/pullbacks.lagda.md]: But `γ` is not assumed to be a pullback

Comment (EgbertRijke) [reply src/foundation-core/pullbacks.lagda.md]: ah, ok.

**Comment by fredrik-bakke** [comment]
Seeing as the conversations have been resolved, I'm merging this pullback. Thanks for the review!

**Pull request merged**